### PR TITLE
[TOPI][batch_matmul] Allow cblas batch_matmul implicit batch_size broadcast

### PR DESCRIPTION
**Pull request by ymwangg**
This PR enables implicit batch_size broadcast in topi.x86.batch_matmul. This is useful for some NLP models.

### CODE REVIEW

File: python/tvm/topi/x86/batch_matmul.py [hunk 1]
```diff
@@ -162,10 +162,10 @@ def batch_matmul_blas_common(cfg, x, y, out_shape, lib):
     assert len(x.shape) == 3 and len(y.shape) == 3, "only support 3-dim batch_matmul"
     XB, M, XK = get_const_tuple(x.shape)
     YB, N, YK = get_const_tuple(y.shape)
-    assert XB == YB, "batch dimension doesn't match"
+    assert (XB == YB) or (YB == 1) or (XB == 1), "batch dimension doesn't match"
```
Comment (comaniac) [inline python/tvm/topi/x86/batch_matmul.py:165]: 1.From this statement it seems the case of `A(1, M, k) x B(8, N, K)` is supported. If so, please add a case to the test.
2. Please update the docstring. The shapes of x and y are no longer just `[batch, ...]`.

File: tests/python/contrib/test_cblas.py [hunk 2]
```diff
@@ -158,10 +158,11 @@ def test_quantized_matmul_add():
 
 
 def verify_batch_matmul(
-    batch, m, l, n, lib, transa=False, transb=False, iterative=False, dtype="float32"
+    batch_a, batch_b, m, l, n, lib, transa=False, transb=False, iterative=False, dtype="float32"
 ):
-    ashape = (batch, l, n) if transa else (batch, n, l)
-    bshape = (batch, m, l) if transb else (batch, l, m)
+    batch = batch_a
```
Comment (comaniac) [inline tests/python/contrib/test_cblas.py:163]: If `batch_a == 1` and `batch_b == B`, then `batch` should be `batch_b`?

**Review by comaniac — CHANGES_REQUESTED** [review]

File: python/tvm/topi/x86/batch_matmul.py (hunk 1 above)
Comment (ymwangg) [reply python/tvm/topi/x86/batch_matmul.py:165]: 1. Added this case. 2 Updated.

File: tests/python/contrib/test_cblas.py (hunk 2 above)
Comment (ymwangg) [reply tests/python/contrib/test_cblas.py]: Fixed. Also changed the extern shape calculation.

**Review by comaniac — APPROVED** [review]
LGTM

**Pull request merged**